Fix broken link at GUIDELINES.md (#569)
metal3-io/metal3-io.github.io: GUIDELINES.md

### Pages

The _[Pages](https://jekyllrb.com/docs/pages/)_ are located at the

> [pages](/) directory, to create one follow these steps:

Reviewer: You can check the Netlify preview site to see if it works as intended.

> [pages](/) directory, to create one follow these steps:

Reviewer: Nevermind, this guideline is not rendered for the website. We also do not have `/pages/` as a construct on the repo. I don't think we have the pages at all anymore, so instead of fixing the links to point to root, I'd think we need to remove the whole Pages section.

/cc @kashifest @lentzi90 @Rozzii WDYT?

> [pages](/) directory, to create one follow these steps:

Author: Yes, it looked odd to me that the guideline was mentioning something that does not exist in the repo! I just tried to add some link that made the most sense but i agree with you. I'll wait on a final call here.

> [pages](.) directory, to create one follow these steps:

Reviewer 2: ```suggestion
[root](.) of the repository, to create one follow these steps:
```

Reviewer: Ah it refers to those pages. Part of those seem to conflict with *.html files in /include etc, so its a not super clear. OK, let's have it with your suggestion.

> [pages](/) directory, to create one follow these steps:

Reviewer: See https://github.com/metal3-io/metal3-io.github.io/pull/569#discussion_r2135176869

> - Create the markdown file, _filename.md_, in [pages](.) directory

Reviewer 2: ```suggestion
- Create the markdown file, _filename.md_, in the [root](.) directory
```
